RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -813,10 +827,23 @@ double AntiElectronIDMVA6::MVAValue(const reco::PFTau& thePFTau)
   Float_t TauPhi = thePFTau.phi();
   float sumPhiTimesEnergy = 0.;
   float sumEnergyPhi = 0.;
-  for ( std::vector<reco::PFCandidatePtr>::const_iterator pfCandidate = signalPFCands.begin();
-	pfCandidate != signalPFCands.end(); ++pfCandidate ) {
-    sumPhiTimesEnergy += (*pfCandidate)->positionAtECALEntrance().phi()*(*pfCandidate)->energy();
-    sumEnergyPhi += (*pfCandidate)->energy();
+  if ( !usePhiAtEcalEntranceExtrapolation ){
+    for ( std::vector<reco::PFCandidatePtr>::const_iterator pfCandidate = signalPFCands.begin();
+	  pfCandidate != signalPFCands.end(); ++pfCandidate ) {
+      sumPhiTimesEnergy += (*pfCandidate)->positionAtECALEntrance().phi()*(*pfCandidate)->energy();
+      sumEnergyPhi += (*pfCandidate)->energy();
+    }
+  }
+  else{
+    TauPhi= -99.;
+    for (unsigned int o = 0; o < signalPFCands.size(); o++ ) {
+      reco::Candidate const*  signalCand = dynamic_cast<reco::Candidate const*> (signalPFCands[o].get());
```

Comment:
same as https://github.com/cms-sw/cmssw/pull/17624#discussion_r104264311
the cast is apparently not needed